File: backend/app/utils/response.py
from typing import Any, Optional
from fastapi.responses import JSONResponse


def success_response(
    data: Any = None,
    message: str = "Success",
    status_code: int = 200,
):
    return JSONResponse(
        status_code=status_code,
        content={
            "success": True,
            "message": message,
            "data": data,
        },
    )


def error_response(
    message: str = "Error",
    status_code: int = 400,
    data: Optional[Any] = None,
):
    return JSONResponse(
        status_code=status_code,
        content={
            "success": False,
            "message": message,
            "data": data,
        },
    )

File: backend/utils/response.js
const successResponse = (res, data = null, message = 'Success', status = 200) => {
  res.status(status).json({
    success: true,
    message,
    data,
  });
};

const errorResponse = (res, message = 'Error', status = 400) => {
  res.status(status).json({
    success: false,
    message,
  });
};

module.exports = { successResponse, errorResponse };
